Doc/library/socket.rst

```diff
@@ -595,6 +595,51 @@ The following functions all create :ref:`socket objects <socket-objects>`.
    .. versionchanged:: 3.2
       *source_address* was added.
 
+.. function:: create_server(address, *, family=AF_INET, backlog=None, reuse_port=False, dualstack_ipv6=False)
```

Comment:
Creating a new API in 2019 that defaults to IPv4 without dualstack seems backwards. People will use these defaults without realizing that they're shutting out a big part of the world.

Our defaults should require users to opt-out of dualstack IPv6 and that the dualstack parameter should perhaps be a tri-state rather than a bool: 'no', 'required', 'allow'.  the default I'd look for would be allow.  such that it would not be an error on a mythical non-dualstack capable host.

When dualstack is enabled, does the family= value have a useful meaning?

On non-dualstack-socket capable hosts (what are those even?) where the user needs a server listening on _both_ IPv4 and IPv6, isn't it true that they need to create _two server sockets_?  `create_server()` doesn't really have a way to address that case.  Supporting such a situation is going to require a `has_dualstack_ipv6()` conditional and wrangling multiple sockets no matter what.

For a situation where people just want a single "most capable" server socket, your four line conditional code seems like it belongs inside of `create_server` itself.